File: frontend/src/types/system-management.ts
export type SystemManagementMasterData = {
  project_count: number;
  center_count: number;
  stage_count: number;
  disabled_stage_count: number;
  stage_template_count: number;
  dictionary_count: number;
  disabled_dictionary_count: number;
};

export type SystemManagementIdentity = {
  user_count: number;
  active_user_count: number;
  inactive_user_count: number;
  role_count: number;
  system_role_count: number;
};

export type SystemManagementAudit = {
  operation_log_count: number;
};

export type SystemManagementWorkflows = {
  pdf_packet_count: number;
  pdf_packet_segment_count: number;
  correction_task_count: number;
  open_correction_task_count: number;
  pending_review_task_count: number;
};

export type SystemManagementManualMaintenance = {
  milestone_count: number;
  enrollment_plan_count: number;
  subject_overview_count: number;
  device_handover_count: number;
  subject_result_count: number;
  clinical_event_count: number;
  device_issue_count: number;
  important_task_count: number;
  total_count: number;
};

export type SystemManagementOverview = {
  master_data: SystemManagementMasterData;
  identity: SystemManagementIdentity;
  audit: SystemManagementAudit;
  workflows: SystemManagementWorkflows;
  manual_maintenance: SystemManagementManualMaintenance;
};

File: backend/app/schemas/system_management.py
from pydantic import BaseModel


class SystemManagementMasterDataRead(BaseModel):
    project_count: int = 0
    center_count: int = 0
    stage_count: int = 0
    disabled_stage_count: int = 0
    stage_template_count: int = 0
    dictionary_count: int = 0
    disabled_dictionary_count: int = 0


class SystemManagementIdentityRead(BaseModel):
    user_count: int = 0
    active_user_count: int = 0
    inactive_user_count: int = 0
    role_count: int = 0
    system_role_count: int = 0


class SystemManagementAuditRead(BaseModel):
    operation_log_count: int = 0


class SystemManagementWorkflowsRead(BaseModel):
    pdf_packet_count: int = 0
    pdf_packet_segment_count: int = 0
    correction_task_count: int = 0
    open_correction_task_count: int = 0
    pending_review_task_count: int = 0


class SystemManagementManualMaintenanceRead(BaseModel):
    milestone_count: int = 0
    enrollment_plan_count: int = 0
    subject_overview_count: int = 0
    device_handover_count: int = 0
    subject_result_count: int = 0
    clinical_event_count: int = 0
    device_issue_count: int = 0
    important_task_count: int = 0
    total_count: int = 0


class SystemManagementOverviewRead(BaseModel):
    master_data: SystemManagementMasterDataRead = SystemManagementMasterDataRead()
    identity: SystemManagementIdentityRead = SystemManagementIdentityRead()
    audit: SystemManagementAuditRead = SystemManagementAuditRead()
    workflows: SystemManagementWorkflowsRead = SystemManagementWorkflowsRead()
    manual_maintenance: SystemManagementManualMaintenanceRead = SystemManagementManualMaintenanceRead()
